pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go

```diff
@@ -73,8 +74,22 @@ func (r *PerconaServerMongoDBRestore) CheckFields() error {
 		return fmt.Errorf("spec clusterName field is empty")
 	}
 
-	if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
-		return errors.New("fields backupName or storageName and destination is empty")
+	if len(r.Spec.BackupName) == 0 && r.Spec.BackupSource == nil {
+		return errors.New("one of backupName or backupSource is required")
```

Comment:
now we have both - r.Spec.StorageName and r.Spec.BackupSource.StorageName
looks like a mistake

Reply:
fixed. now we're using ` r.Spec.StorageName` everywhere